arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -217,6 +217,10 @@ public VectorHolder read(VectorHolder reuse, int numValsToRead) {
   }
 
   private void allocateFieldVector(boolean dictionaryEncodedVector) {
+    // Allocate-only: caller must ensure there is no active vector in use.
+    Preconditions.checkState(
+        vec == null,
+        "allocateFieldVector must be called only when no active vector is in use (vec == null)");
```

Comment:
nit: typically error msg is more user friendly and doesn't include method or variable names. maybe sth like?
```
Allocation must be called only when previous vector instance was released
```

Reply:
We could check if the vec.isClosed? I think that also is fine and let garbage collection just drag it away